packages/core/src/components/DataTable/index.tsx

```diff
@@ -0,0 +1,437 @@
+  sortedDataList: IndexedParentRow[];
+  expandedRows: Set<number>;
+  selectedRows: SelectedRows;
+  sortBy: string;
+  sortDirection: SortDirectionType;
+  editMode: boolean;
+};
+
+/** A dynamic and responsive table for displaying tabular data. */
+export class DataTable extends React.Component<DataTableProps & WithStylesProps, State> {
+  state = {
+    changeLog: {},
+    preEditSortedDataList: [],
+    sortedDataList: indexDataList(this.props.data!),
+    expandedRows: new Set(),
+    selectedRows: {},
+    sortBy: this.props.sortByOverride || '',
+    sortDirection: this.props.sortDirectionOverride!,
+    editMode: false,
+  };
+
+  static defaultProps: Pick<DataTableProps, DefaultDataTableProps> = {
+    columnHeaderHeight: undefined,
+    columnMetadata: {},
+    columnToLabel: {},
+    data: [],
+    defaultEditCallback: () => {},
+    editable: false,
+    editCallbacks: {},
+    enactEditsCallback: () => {},
+    expandable: false,
+    extraHeaderButtons: [],
+    height: 400,
+    instantEdit: true,
+    keys: [],
+    renderers: {},
+    rowHeight: 'regular',
+    selectable: false,
+    selectOnRowClick: false,
+    showColumnDividers: false,
+    showRowDividers: false,
+    sortByOverride: '',
+    sortCallback: () => {},
+    sortDirectionOverride: SortDirection.ASC,
+    sortOverride: false,
+    tableHeaderHeight: undefined,
+    tableHeaderLabel: '',
+    width: 0,
+    zebra: false,
+  };
+
+  // Infers keys from data if they aren't explicitely defined
+  keys =
+    this.props.keys && this.props.keys.length > 0
+      ? this.props.keys
+      : Array.from(
+          this.props.data!.reduce((keySet: Set<string>, row: ParentRow) => {
+            Object.keys(row.data).forEach(key => {
+              if (row.metadata === undefined || row.metadata.colSpanKey !== key) {
+                keySet.add(key);
+              }
+            });
+
+            return keySet;
+          }, new Set()),
+        );
+
+  rowStyles = (expandedDataList: ExpandedRow[]) => ({ index }: { index: number }): RowStyles => ({
+    background: getRowColor(
+      expandedDataList[index],
+      index,
+      this.props.zebra || false,
+      this.props.theme,
+    ),
+    display: 'flex',
+    flexDirection: 'row',
+    alignItems: 'center',
+    borderBottom: this.props.showRowDividers ? '1px solid' : '',
+    borderColor: this.props.theme!.color.core.neutral[1],
+    outline: 'none',
+  });
+
+  private sort = ({
+    sortBy,
+    sortDirection,
+  }: {
+    sortBy: string;
+    sortDirection: SortDirectionType;
+  }): void => {
+    const { sortOverride, sortCallback } = this.props;
+    if (sortOverride && sortCallback) {
+      sortCallback(sortBy, sortDirection);
+    } else {
+      const { sortedDataList } = this.state;
+
+      this.setState({
+        sortedDataList: sortList(sortedDataList, this.keys, sortBy, sortDirection),
+        sortBy,
+        sortDirection,
+      });
+    }
+  };
+
+  // eslint-disable-next-line
```

Comment:
What is this disabling?

Reply:
There's a weird prettier error that insists on invalid formatting, without this I get
```use a line break```
But doing that results in
```Insert `·(`   ```
Which leaves me with
```Replace `(event:·React.SyntheticEvent<EventTarget>` with `event:·React.SyntheticEvent<EventTarget>,⏎··````
Which leaves me with a hanging ```=>``` which doesn't make any sense